### ISSUE TITLE
[Feature Request]

### ISSUE THREAD

**Issue by brson** [solana-labs/move#33]
Switch to the sbf-solana-solana target.

Per comments here https://github.com/solana-labs/move/pull/24#discussion_r1096844747 the solana tooling has already switched the the sbf target, so our code should use that instead.

I think this will just involve changing target triples to `sbf-solana-solana` and using the sbf tools instead of the bpf tools, changing the `BPF_TOOLS_ROOT` variable to `SBF_TOOLS_ROOT`.

There are some llvm features, I think `+sbfv1` and `+sbfv2`, that I don't understand yet. Hopefully though the sbf target will turn on `sbfv1` and `sbfv2` is opt-in.

**Comment by brson** [comment]
It looks like `sbfv2`  is the name of the CPU in llvm. Hopefully if we don't specify anything about the cpu for the target, we'll just get sbfv1.

**Comment by brson** [comment]
Oh, right now we're specifying the "generic" during codegen, that's probably where we can say "sbfv1" or "sbfv2".

**Comment by nvjle** [comment]
Yes. Specifying target `-bpfel` with feature `+solana` is essentially the same as specifying target `-sbf-solana-solana`. Beyond that, `-mcpu` will work the same either way (today), and you must specifically ask for `sbfv2` if you want it. See also `Target/SBF/SBF.td` for the exact spellings.
